### PR TITLE
Add jobs for Drone MacOS/ARM exec runners

### PR DESCRIPTION
**Pull request by webvictim**
Adds jobs for:
- Building teleport .tar.gz on MacOS
- Building teleport .pkg on MacOS
- Building signed/notarized tsh .pkg on MacOS
- Building teleport .tar.gz on ARMv7

Drone runner setup is handled by https://github.com/gravitational/ops/pull/164

Ticks off more issues in https://github.com/gravitational/teleport/issues/3936

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1558,6 +1558,450 @@ volumes:
...
+platform:
+  os: darwin
+  arch: amd64
+
+# trigger:
+#   event:
+#     - tag
+#   ref:
+#     include:
+#       - refs/tags/v*
```
Comment (russjones) [inline .drone.yml]: Remove comments.

File: .drone.yml [hunk 2]
```diff
@@ -1558,6 +1558,450 @@ volumes:
...
```
Comment (russjones) [inline .drone.yml]: Remove comments.

File: .drone.yml [hunk 3]
```diff
@@ -1558,6 +1558,450 @@ volumes:
...
```
Comment (russjones) [inline .drone.yml]: Remove comments.

File: Makefile [hunk 4]
```diff
@@ -41,10 +41,21 @@ endif
...
+endif
+
+# PAM headers for Darwin live under /usr/local/include/security instead, as SIP
+# prevents us from modifying/creating /usr/include/security on newer versions of MacOS
+ifeq ("$(OS)","darwin")
+ifneq ("$(wildcard /usr/local/include/security/pam_appl.h)","")
+PAM_TAG := pam
+PAM_MESSAGE := "with PAM support"
+endif
+endif
```
Comment (russjones) [inline Makefile:58]: Why not expand the `ifeq` statements above instead of making this another block?

File: .drone.yml (hunk 1 above)
Comment (webvictim) [reply .drone.yml]: This is already done - bit of a race condition where I marked this ready for review, but then had to go back and comment some stuff for additional testing.

File: Makefile (hunk 4 above)
Comment (webvictim) [reply Makefile:58]: Maybe I'm being dense, but what do you mean by this? I think I could rewrite it this way to drop the Darwin check (as it looks like cgo includes `/usr/include` and `/usr/local/include` by default):

```make
# PAM support will only be built into Teleport if headers exist at build time.
PAM_MESSAGE := "without PAM support"
ifneq ("$(wildcard /usr/include/security/pam_appl.h)","")
PAM_TAG := pam
PAM_MESSAGE := "with PAM support"
else
# PAM headers for Darwin live under /usr/local/include/security instead, as SIP
# prevents us from modifying/creating /usr/include/security on newer versions of MacOS
ifneq ("$(wildcard /usr/local/include/security/pam_appl.h)","")
PAM_TAG := pam
PAM_MESSAGE := "with PAM support"
endif
endif
```

I'm not sure that's any much clearer/simpler though.

Comment (webvictim) [reply Makefile]: (I'm pretty sure `ifeq` and `ifneq` only support one condition for a test)

Comment (russjones) [reply Makefile]: Okay, makes sense.

**Review by russjones — APPROVED** [review]

File: .drone.yml [hunk 5]
```diff
@@ -1558,6 +1558,435 @@ volumes:
...
+
+  - name: Clean up exec runner storage
+    commands:
+      - chmod -R u+rw /tmp/build-darwin-amd64-pkg
+      - rm -rf /tmp/build-darwin-amd64-pkg/go
+
+---
+kind: pipeline
+type: exec
+name: build-darwin-amd64-pkg-tsh
```
Comment (awly) [inline .drone.yml:1771]: can this be combined with the above pipeline?
they both seem to be building darwin-amd64 .pkg files

File: .drone.yml [hunk 6]
```diff
@@ -1558,6 +1558,435 @@ volumes:
...
+      - export VERSION=$(cat /tmp/build-darwin-amd64-pkg/go/.version.txt)
+      - make pkg OS=$OS ARCH=$ARCH
+
+  - name: Copy Mac pkg artifacts
+    commands:
+      - cd /tmp/build-darwin-amd64-pkg/go/src/github.com/gravitational/teleport
+      # delete temporary tarball artifacts so we don't re-upload them in the next stage
+      - rm -rf /tmp/build-darwin-amd64-pkg/go/artifacts/*.tar.gz
+      # copy release archives to artifact directory
+      - find build -maxdepth 1 -iname "teleport*.pkg" -print -exec cp {} /tmp/build-darwin-amd64-pkg/go/artifacts \;
```
Comment (awly) [inline .drone.yml:1745]: Can this be `cp build/teleport*.pkg /tmp/build-darwin-amd64-pkg/go/artifacts` ?

Comment (webvictim) [reply .drone.yml:1745]: Yes, I guess it could.

File: .drone.yml (hunk 5 above)
Comment (webvictim) [reply .drone.yml:1771]: I think I'd prefer to keep each artifact in its own separate pipeline for parallelisation. I know that it's more boilerplate YAML, but the idea should be that each artifact will either build or fail completely separately.

File: .drone.yml (hunk 6 above)
Comment (webvictim) [reply .drone.yml]: Fixed.

**Review by awly — APPROVED** [review]

**Pull request merged**